File: 2125_number_of_laser_beams_in_a_bank.py
class Solution:
    def numberOfBeams(self, bank: list[str]) -> int:
        num_beams = 0
        prev_count: int | None = None

        for floor in bank:
            count = 0
            for c in floor:
                if c == "1":
                    count += 1
            if prev_count is not None:
                num_beams += count * prev_count
            if count > 0:
                prev_count = count

        return num_beams

File: problems/2125-number-of-laser-beams-in-a-bank.ts
function numberOfBeams(bank: string[]): number {
  let numBeams = 0
  let prevCount: number | undefined = undefined

  for (const floor of bank) {
    let count = 0
    for (const c of floor) {
      if (c === "1") {
        count++
      }
    }
    if (prevCount !== undefined) {
      numBeams += prevCount * count
    }
    if (count !== 0) {
      prevCount = count
    }
  }
  return numBeams
}
